include/ddc/discrete_element.hpp

```diff
@@ -81,24 +82,44 @@ KOKKOS_FUNCTION constexpr DiscreteElement<QueryTags...> select(
     return DiscreteElement<QueryTags...>(std::move(arr));
 }
 
-template <class QueryTag, class HeadTag, class... TailTags>
-KOKKOS_FUNCTION constexpr DiscreteElement<QueryTag> const& take(
-        DiscreteElement<HeadTag> const& head,
-        DiscreteElement<TailTags> const&... tags)
+template <class QueryTag, class... DE>
+struct Contains;
+
+template <class QueryTag>
+struct Contains<QueryTag> : std::false_type
+{
+};
+
+template <class QueryTag, class... HeadTags, class... DETail>
+struct Contains<QueryTag, DiscreteElement<HeadTags...>, DETail...>
+    : std::conditional<
+              type_seq_contains_v<detail::TypeSeq<QueryTag>, detail::TypeSeq<HeadTags...>>,
+              std::true_type,
+              typename Contains<QueryTag, DETail...>::type>
+{
+};
+
+template <class QueryTag, class... DE>
+using contains_t = typename Contains<QueryTag, DE...>::type;
+
+template <class QueryTag, class... DE>
+inline constexpr bool contains_v = contains_t<QueryTag, DE...>::value;
```

Comment:
I think we can develop a conversion tool between `DiscreteElement`/`DiscreteVector` and `TypeSeq` to reuse more what is already available in `TypeSeq`